games/game_stalkeranomaly.py

```diff
@@ -132,20 +141,724 @@ def getContentsFor(self, tree: mobase.IFileTree) -> List[int]:
+        vec.x = self.clamp(vec.x, fmin.x, fmax.x)
+        vec.y = self.clamp(vec.x, fmin.y, fmax.y)
+        vec.z = self.clamp(vec.z, fmin.z, fmax.z)
+
+    def fqt_q8(self, reader: IReader):
+        vec = IVec4(0.0, 0.0, 0.0, 0.0)
+        vec.x = self.f_q8(reader, -1.0, 1.0)
+        vec.y = self.f_q8(reader, -1.0, 1.0)
+        vec.z = self.f_q8(reader, -1.0, 1.0)
+        vec.w = self.f_q8(reader, -1.0, 1.0)
+        vec.x = self.clamp(vec.x, -1.0, 1.0)
+        vec.y = self.clamp(vec.y, -1.0, 1.0)
+        vec.z = self.clamp(vec.z, -1.0, 1.0)
+        vec.w = self.clamp(vec.w, -1.0, 1.0)
+
+    def f_q8(self, reader: IReader, fmin: float, fmax: float):
+        return (float(reader.u8()) / 255.0) * (fmax - fmin) + fmin
+
+
+class XRBoneData:
+    def __init__(self):
+        self.bones_mask = -1
+        self.root_bone = 0
+        self.min = IVec3(0.0, 0.0, 0.0)
+        self.max = IVec3(0.0, 0.0, 0.0)
+        self.bones = []
+
+    def load(self, reader: IReader):
+        self.bones_mask = reader.u64()
+        self.root_bone = reader.u16()
+        self.min = reader.fvec3()
+        self.max = reader.fvec3()
+        bones_count = reader.u16()
+        for x in range(bones_count):
+            bone = XRNETState()
+            bone.read(reader, self.min, self.max)
+            self.bones.append(bone)
+
+
+class XRSkeleton:
+    def __init__(self):
+        self.source_id = -1
+        self.saved_bones = XRBoneData()
+
+    def read_state(self, reader: IReader):
+        self.visual_animation = reader.str()
+        flags = IFlag(reader.u8())
+        self.source_id = reader.u16()
+        if flags.has(4):
+            self.saved_bones.load(reader)
+
+
+class XRObject(XRAbstract):
+    def __init__(self):
+        super().__init__()
+        self.graph_id = 0
+        self.distance = 0.0
+        self.direct_control = True
+        self.node_id = 0
+        self.story_id = -1
+        self.spawn_story_id = -1
+
+    def read_spawn(self, reader: IReader):
+        super().read_spawn(reader)
+        size = reader.u16()
+        if size <= 2:
+            self._valid = False
+            return
+        self.read_state(reader)
+
+    def read_state(self, reader: IReader):
+        self.graph_id = reader.u16()
+        self.distance = reader.float()
+        self.direct_control = bool(reader.u32())
+        self.node_id = reader.u32()
+        self.flags.set(reader.u32())
+        self.ini_str = reader.str()
+        self.story_id = reader.u32()
+        self.spawn_story_id = reader.u32()
+
+    def read_update(self, reader: IReader):
+        update = reader.u16()
+        if update != XRFlag.MSG_UPDATE:
+            self._valid = False
+            return
+
+
+class XRDynamicObject(XRObject):
+    def __init__(self):
+        super().__init__()
+        self.time_id = -1
+        self.switch_counter = 0
+
+
+class XRDynamicObjectVisual(XRDynamicObject, XRVisual):
+    def __init__(self):
```

Comment:
This init is probably not necessary, this is defaulted by Python. The only place where you'd need is when extending Boost.Python wrapped classes because those are not working properly.